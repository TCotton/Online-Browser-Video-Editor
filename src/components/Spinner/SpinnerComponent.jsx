import React from 'react'
import { useSelector } from 'react-redux'
import { displayLoader } from '../slices/loaderSlice'

export const SpinnerComponent = () => {
  const display = useSelector(displayLoader)

  return display ? (
    <div data-testid='spinner'>
      <div className='lds-dual-ring' />
    </div>
  ) : null
}
