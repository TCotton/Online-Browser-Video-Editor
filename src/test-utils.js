import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore, combineReducers } from 'redux'
import { Provider } from 'react-redux'

import counterReducer from './components/slices/counterSlice';
import filesReducer from './components/slices/filesSlice';
import videoReducer from './components/slices/videoSlice';
import playerReducer from './components/slices/playerSlice';
import audioReducer from './components/slices/audioSlice';
import imageReducer from './components/slices/imageSlice';
import waveformReducer from './components/slices/waveformSlice';

function render(
    ui,
    {
        initialState,
        store = createStore(combineReducers({
            counter: counterReducer,
            files:  filesReducer,
            video: videoReducer,
            player: playerReducer,
            audio: audioReducer,
            images: imageReducer,
            waveform: waveformReducer,
        }), initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react'
// override render method
export { render }