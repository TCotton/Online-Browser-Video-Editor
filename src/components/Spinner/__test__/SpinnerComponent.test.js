import React from 'react'
import { SpinnerComponent } from '../SpinnerComponent'
import { render } from '../../../test-utils'

describe('SpinnerComponent', () => {
  it('should render component', function () {
    const { container } = render(<SpinnerComponent />, {
      initialState: {
        loader: {
          display: true
        }
      }
    })
    expect(container.querySelector('[data-testid="spinner"]')).not.toBeNull()
  })

  it('should match snapshot', () => {
    const { baseElement } = render(<SpinnerComponent />, {
      initialState: {
        loader: {
          display: true
        }
      }
    })
    expect(baseElement).toMatchSnapshot()
  })
})
