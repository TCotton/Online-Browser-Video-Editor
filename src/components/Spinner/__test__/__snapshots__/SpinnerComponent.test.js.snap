// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerComponent should match snapshot 1`] = `
<body>
  <div>
    <div
      data-testid="spinner"
    >
      <div
        class="lds-dual-ring"
      />
    </div>
  </div>
</body>
`;
